api/Service/EventService.cs

```diff
@@ -8,119 +8,119 @@
+                ContactEmail = eventDto.ContactEmail,
+                ContactPhone = eventDto.ContactPhone,
+                Tags = eventDto.Tags,
+                BackgroundId = await _pictureService.CreatePictureByContent(eventDto.BackgroundImage)
+            };
+            return newEvent;
+        }
 
-    public async Task<Event> GetEvent(int id)
-    {
-        var myEvent = await _context.Events.Include(a => a.EventLocation).FirstOrDefaultAsync(a => a.Id == id);
-        return myEvent;
-    }
+        public async Task<bool> DeleteEvent(int eventID)
+        {
+            var myEvent = await _context.Events.FindAsync(eventID);
+            if(myEvent == null) {
+                return false;
+            }
+            _context.Events.Remove(myEvent);
+            return await _context.SaveChangesAsync() > 0;
+        }
 
-    public async Task<Event> PostEvent(Event newEvent)
-    {
-        newEvent.EventLocation = await _locationService.PostLocation(newEvent.EventLocation);
-        _context.Events.Add(newEvent);
-        await _context.SaveChangesAsync();
-        return newEvent;
-    }
+        public async Task<IList<Event>> GetEventList()
+        {
+            var events = await _context.Events.Include(a => a.EventLocation).ToListAsync();
+            return events;
+        }
 
-    public async Task<Event> PutEvent(Event eventToPut)
-    {
-        var myEvent = await _context.Events.FindAsync(eventToPut.Id);
-        if(myEvent == null) {
-            return null;
+        public async Task<Event> GetEvent(int id)
+        {
+            var myEvent = await _context.Events.Include(a => a.EventLocation).FirstOrDefaultAsync(a => a.Id == id);
+            return myEvent;
         }
-        myEvent.CategoryId = eventToPut.CategoryId;
-        myEvent.HostId = eventToPut.HostId;
-        myEvent.Title = eventToPut.Title;
-        myEvent.StartingDate = eventToPut.StartingDate;
-        myEvent.EndingDate = eventToPut.EndingDate;
-        myEvent.MinimumParticipants = eventToPut.MinimumParticipants;
-        myEvent.MaximumParticipants = eventToPut.MaximumParticipants;
-        myEvent.AutoCancel = eventToPut.AutoCancel;
-        myEvent.AutoJoin = eventToPut.AutoJoin;
-        myEvent.JoinDeadline = eventToPut.JoinDeadline;
-        myEvent.Fee = eventToPut.Fee;
-        myEvent.Description = eventToPut.Description;
-        myEvent.EventLocation = await _locationService.PostLocation(eventToPut.EventLocation);   
-        myEvent.ContactEmail = eventToPut.ContactEmail;
-        myEvent.ContactPhone = eventToPut.ContactPhone;
-        myEvent.Tags = eventToPut.Tags;
-        myEvent.BackgroundId = eventToPut.BackgroundId;
 
-        _context.Events.Update(myEvent);
-        await _context.SaveChangesAsync();
-        return myEvent;
-    }
+        public async Task<Event> PostEvent(Event newEvent)
+        {
+            newEvent.EventLocation = await _locationService.PostLocation(newEvent.EventLocation);
```

Comment:
No. In the locationService in PostLocation it firstly checks if a location exists. If it exists it returns me the existing location with the correct id. otherwise it posts it in the database with a new id and returns the posted location